include/fuzzuf/algorithms/afl/templates/afl_mutation_hierarflow_routines.hpp

```diff
@@ -17,536 +17,549 @@
+    /* Copy head */
+    if (i != mutator.GetLen()) {
+      ex_tmp[i] = mutator.GetBuf()[i];
+    }
+  }
 
-    state.stage_finds[option::STAGE_EXTRAS_UI]  += new_hit_cnt - orig_hit_cnt;
-    state.stage_cycles[option::STAGE_EXTRAS_UI] += state.stage_max;
+  u64 new_hit_cnt = state.queued_paths + state.unique_crashes;
 
-    return this->GoToDefaultNext();
+  state.stage_finds[option::STAGE_EXTRAS_UI] += new_hit_cnt - orig_hit_cnt;
+  state.stage_cycles[option::STAGE_EXTRAS_UI] += state.stage_max;
+
+  return this->GoToDefaultNext();
 }
 
-template<class State>
-HavocBaseTemplate<State>::HavocBaseTemplate(State &state)
-    : state(state) {}
+template <class State>
+HavocBaseTemplate<State>::HavocBaseTemplate(State &state) : state(state) {}
 
 // Because the process details of Havoc will be used also in Splicing,
-// we extract the "core" of Havoc into another HierarFlowRoutine, HavocBaseTemplate.
-// both Havoc and Splicing will inherit this and use DoHavoc
-template<class State>
-template<typename CustomCases>
+// we extract the "core" of Havoc into another HierarFlowRoutine,
+// HavocBaseTemplate. both Havoc and Splicing will inherit this and use DoHavoc
+template <class State>
+template <typename CustomCases>
 bool HavocBaseTemplate<State>::DoHavoc(
-    AFLMutatorTemplate<State>& mutator,
-    optimizer::Optimizer<u32> &mutop_optimizer,
-    CustomCases custom_cases,
-    const std::string &stage_name,
-    const std::string &stage_short,
+    AFLMutatorTemplate<State> &mutator,
+    optimizer::Optimizer<u32> &mutop_optimizer, CustomCases custom_cases,
+    const std::string &stage_name, const std::string &stage_short,
     u32 perf_score,
-    s32 stage_max_multiplier, // see directly below
-    int stage_idx
-) {
-    state.stage_name = stage_name;
-    state.stage_short = stage_short;
-    state.stage_max = stage_max_multiplier * perf_score / state.havoc_div / 100;
-    state.stage_cur_byte = -1;
-
-    if (state.stage_max < option::GetHavocMin(state)) {
-        state.stage_max = option::GetHavocMin(state);
-    }
+    s32 stage_max_multiplier,  // see directly below
+    int stage_idx) {
+  state.stage_name = stage_name;
+  state.stage_short = stage_short;
+  state.stage_max = stage_max_multiplier * perf_score / state.havoc_div / 100;
+  state.stage_cur_byte = -1;
 
-    u64 orig_hit_cnt = state.queued_paths + state.unique_crashes;
+  if (state.stage_max < option::GetHavocMin(state)) {
+    state.stage_max = option::GetHavocMin(state);
+  }
 
-    u64 havoc_queued = state.queued_paths;
+  u64 orig_hit_cnt = state.queued_paths + state.unique_crashes;
 
-    /* We essentially just do several thousand runs (depending on perf_score)
-       where we take the input file and make random stacked tweaks. */
+  u64 havoc_queued = state.queued_paths;
 
-    for (state.stage_cur = 0; state.stage_cur < state.stage_max; state.stage_cur++) {
-        using afl::util::UR;
+  std::array<u64, fuzzuf::mutator::NUM_CASE> havoc_operator_finds;
```

Comment:
I think this is not initialized and hence causes Use of Uninitialized Value.

Reply:
Maybe you should check the other files where std::array is used (e.g. MOptParticle::fitness).

Reply:
fixed in 3aed518b55a335259bd324e710f4d0bac862ddef

Reply:
and fixed uninitialized things in 17e719804bfb4cf61b175ef093e608febae6b240